src/db/collections/node_configuration.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use mongodb::{bson::doc, error::Error};
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    db::{
+        mongodb::{MongoDbCollection, MongoDbCollectionExt},
+        MongoDb,
+    },
+    types::node::NodeConfiguration,
+};
+
+/// The corresponding MongoDb document representation to store [`NodeConfiguration`]s.
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct NodeConfigurationDocument {
+    pub config: NodeConfiguration,
+}
```

Comment:
Why not use a tuple struct as wrapper? We shouldn't have the indirection through `config` in the database. Alternatively we could use `#[serde(flatten)]`.

Reply:
I look into it.

Reply:
Decided to go with the new-type solution.